xcube/webapi/ows/coverages/controllers.py

```diff
@@ -191,9 +473,17 @@ def dataset_to_image(
             if ds[var].dtype not in {np.uint8, np.uint16}:
                 ds[var] = ds[var].astype(np.uint16, casting='unsafe')
 
+    ds = ds.squeeze()
+
     with tempfile.TemporaryDirectory() as tempdir:
         path = os.path.join(tempdir, 'out.' + image_format)
-        ds.rio.to_raster(path)
+        ds = ds.drop_vars(
+            names=['crs', 'spatial_ref'], errors='ignore'
+        ).squeeze()
```

Comment:
Why is this necessary?

Reply:
The conversion to TIFF/PNG doesn't work if there are extra variables or dimensions. I've added a comment mentioning this.